Orange-OpenSource/its-client
# rust: switch project layout from workspace to lib

What's new
---------------

The main purpose of this crate is to provide a library to create ITS message production/consumption clients, not to provide clients
Through the [2.0.0 refactoring][#93] the idea is to use a more common project layout by examples aside the library

```
.
├─ Cargo.toml
├─ src/
│  ├─ lib.rs
│  ├─ ...
│  ...
└─ examples/
   ├─ copycat.rs
   ...
```

This pull request initiates the refactoring by only applying the layout changes with no code in it yet
Refactored code and examples will be added later

Old code is still present in order to avoid changes history loss as much as possible (and of course to ease refactoring by having the old code right away)
      
How to test
---------------

Nothing much than checking pipelines are ok
You can check cargo build by yourself if interested

```
cargo clean; cargo build --all; cargo test --all; cargo clippy -- -D clippy::all
```
